meilisearch-core/src/update/documents_addition.rs

```diff
@@ -143,6 +163,12 @@ pub fn apply_documents_addition<'a, 'b>(
     };
 
     // 3. index the documents fields in the stores
+    // add facets for docs facets TODO move in a function
+    if let Some(attributes_for_facetting) = index.main.attributes_for_faceting(writer)? {
+        let facet_map = facets::facet_map_from_docs(&schema, &documents_additions, attributes_for_facetting.as_ref())?;
+        index.facets.add(writer, facet_map)?;
+    }
```

Comment:
If I understand that part correctly, the `facet_map_from_docs` function retrieves all the facet names/values from the list of `IndexMap` representing the documents?

Reply:
exactly, i can then use this map to either add or remove documents